File: python_files/data_structures/LinkedList.py
class Node:
  def __init__(self, data):
    self.data=data
    self.next=None
  
class LinkedList:
  def __init__(self):
    self.head = None
    self.tail = None
  
  def addNode(self, data):
    if self.head == None:
      self.head = self.tail = Node(data)
      return
    # else
    new_node = Node(data)
    self.tail.next = new_node
    self.tail = self.tail.next
  def getData(self):
    curr = self.head
    if curr!=None:
      while curr != None:
        print(curr.data)
        curr = curr.next

File: c++/Data-Structures/LinkedList.cpp
# include <iostream>

using namespace std;
class Node{
    public:
        int value;
        Node *next;

        Node(int value){
            this->value=value;
            this->next=NULL;
        }
};
class LinkedList{
    
    public:
        Node* head;
        LinkedList(){
            this->head = NULL;
        }

        void insert_node(int value){
            Node * newNode = new Node(value);

            if(this->head == NULL){
                this-> head = newNode;
            }
            else{
                Node * temp = head;
                while(temp->next!=NULL){
                    temp = temp->next;
                }
                temp->next = newNode;
            }
        }

        void print_list(){
            Node * temp = head;
            while (temp!=NULL){
                cout << temp->value << endl;
                temp = temp->next;
            }
        }
};

int main(){
    LinkedList lList = LinkedList();
    lList.insert_node(1);
    lList.insert_node(2);
    lList.insert_node(3);

    lList.print_list();
    cout << endl;
    cout << "head: " << lList.head->value << endl;

    // cout << lList.head->value;

    return 0;
}